pkg/resource/parameter_group/hooks.go

```diff
@@ -154,3 +155,122 @@ func (rm *resourceManager) resetAllParameters(
 	}
 	return &resource{ko}, nil
 }
+
+// getTags gets tags from given ParameterGroup.
+func (rm *resourceManager) getTags(
+	ctx context.Context,
+	resourceARN string,
+) ([]*svcapitypes.Tag, error) {
+	resp, err := rm.sdkapi.ListTagsWithContext(
+		ctx,
+		&svcsdk.ListTagsInput{
+			ResourceArn: &resourceARN,
+		},
+	)
+	rm.metrics.RecordAPICall("GET", "ListTags", err)
+	if err != nil {
+		return nil, err
+	}
+	tags := make([]*svcapitypes.Tag, 0, len(resp.TagList))
+	for _, tag := range resp.TagList {
+		tags = append(tags, &svcapitypes.Tag{
+			Key:   tag.Key,
+			Value: tag.Value,
+		})
+	}
+	return tags, nil
+}
+
+// updateTags updates tags of given ParameterGroup to desired tags.
+func (rm *resourceManager) updateTags(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncTags")
+	defer func() { exit(err) }()
+
+	arn := (*string)(latest.ko.Status.ACKResourceMetadata.ARN)
+
+	toAdd, toDelete := computeTagsDelta(
+		desired.ko.Spec.Tags, latest.ko.Spec.Tags,
+	)
+
+	if len(toDelete) > 0 {
+		rlog.Debug("removing tags from parameter group", "tags", toDelete)
+		_, err = rm.sdkapi.UntagResourceWithContext(
+			ctx,
+			&svcsdk.UntagResourceInput{
+				ResourceArn: arn,
+				TagKeys:     toDelete,
+			},
+		)
+		rm.metrics.RecordAPICall("UPDATE", "UnTagResource", err)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(toAdd) > 0 {
+		rlog.Debug("adding tags to parameter group", "tags", toAdd)
+		_, err = rm.sdkapi.TagResourceWithContext(
+			ctx,
+			&svcsdk.TagResourceInput{
+				ResourceArn: arn,
+				Tags:        sdkTagsFromResourceTags(toAdd),
+			},
+		)
+		rm.metrics.RecordAPICall("UPDATE", "TagResource", err)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func computeTagsDelta(
+	desired []*svcapitypes.Tag,
+	latest []*svcapitypes.Tag,
+) (added []*svcapitypes.Tag, removed []*string) {
+	toDelete := []*string{}
+	toAdd := []*svcapitypes.Tag{}
+
+	desiredTags := map[string]string{}
+	key := ""
+	value := ""
+	for _, tag := range desired {
+		if tag.Key != nil {
+			key = *tag.Key
+		}
+		if tag.Value != nil {
+			value = *tag.Value
+		}
+		desiredTags[key] = value
```

Comment:
if `tag.Key` is `nil` during an iteration of the `for` loop, then the `key` will have value from earlier iteration of the loop, and it will overwrite the value in the `desiredTags` for that.

Consider rewriting the logic as:
```
for _, tag := range desired {
    if (tag.Key != nil && tag.Value != nil) {
      desiredTags[*tag.Key] = *tag.Value
    }
}
```